packages/down-up-mixin/test/down-up-mixin.test.ts

```diff
@@ -0,0 +1,184 @@
+import { LitElement, html, customElement } from 'lit-element';
+import { fixture } from '@open-wc/testing-helpers';
+import { mousedown, mouseup, touchstart, touchend } from '@vaadin/test-helpers';
+import { DownUpMixin } from '../down-up-mixin';
+
+const { expect } = chai;
+const { sinon } = window;
+
+@customElement('dum-element')
+class DumElement extends DownUpMixin(LitElement) {
+  render() {
+    return html`
+      <div>Content</div>
+    `;
+  }
+
+  protected _onDown() {
+    this.dispatchEvent(new CustomEvent('down'));
+  }
+
+  protected _onUp() {
+    this.dispatchEvent(new CustomEvent('up'));
+  }
+}
+
+@customElement('dum-child-element')
+class DumChildElement extends DownUpMixin(LitElement) {
+  render() {
+    return html`
+      <div
+        @mousedown="${this._onMouseDown}"
+        @touchstart="${this._onTouchStart}"
+        @touchend="${this._onTouchEnd}"
+      >
+        Content
+      </div>
+    `;
+  }
+
+  protected get _downUpTarget() {
+    return null;
+  }
+
+  protected _onDown() {
+    this.dispatchEvent(new CustomEvent('down'));
+  }
+
+  protected _onUp() {
+    this.dispatchEvent(new CustomEvent('up'));
+  }
+}
+
+@customElement('dum-multi-element')
+class DumMultiElement extends DownUpMixin(DownUpMixin(LitElement)) {
+  render() {
+    return html`
+      <div>Content</div>
+    `;
+  }
+
+  protected _onDown() {
+    this.dispatchEvent(new CustomEvent('down'));
+  }
+
+  protected _onUp() {
+    this.dispatchEvent(new CustomEvent('up'));
+  }
+}
+
+describe('DownUpMixin', () => {
+  describe('host as target', () => {
+    let element: DumElement;
+    let spy: sinon.SinonSpy;
+
+    beforeEach(async () => {
+      element = await fixture(`<dum-element></dum-element>`);
+      spy = sinon.spy();
+    });
+
+    it('should call down handler on mousedown', () => {
+      element.addEventListener('down', spy);
+      mousedown(element);
+      expect(spy).to.be.calledOnce;
+    });
+
+    it('should call up handler on mouseup after mousedown', () => {
+      element.addEventListener('up', spy);
+      mousedown(element);
+      mouseup(element);
+      expect(spy).to.be.calledOnce;
+    });
+
+    it('should not call down handler on right button mousedown', () => {
```

Comment:
Missing event listener:
```suggestion
    it('should not call down handler on right button mousedown', () => {
      element.addEventListener('down', spy);
```

Reply:
Thanks, done.